blockchain/serviceMetadata.go

```diff
@@ -58,7 +58,14 @@ func ServiceMetaData() *ServiceMetadata {
 	} else {
 		//TO DO, have a snetd command to create a default metadata json file, for now just read from a local file
 		// when block chain reading is disabled
-		metadata, err = readServiceMetaDataFromLocalFile("service_metadata.json")
+		if metadata, err = readServiceMetaDataFromLocalFile("service_metadata.json"); err != nil {
```

Comment:
I would move condition on loading correct metadata: from file or from blockchain to the `components.ServiceMetaData()`  method. 
In pseudocode:
```go
func (components *Components) ServiceMetaData() *blockchain.ServiceMetadata {
	if components.serviceMetadata != nil {
		return components.serviceMetadata
	}
        if len(config.GetString(config.ServiceMetadataFileKey)) > 0 {
		components.serviceMetadata = readServiceMetaDataFromLocalFile("service_metadata.json")
		return components.serviceMetadata
        }
        if !components.Blockchain().Enabled() {
            components.serviceMetadata = blockchain.ServiceMetaData()
            return components.serviceMetadata
        }
        log.Panic("Blockchain is disabled and no service metadata file is passed in configuration")
}
```

Reply:
In pseudocode: looks like a typo in the commented line ( this will go on a infinite recursion)
```go
       if !config.GetBool(config.BlockchainEnabledKey) {
           //components.serviceMetadata = blockchain.ServiceMetaData()
             components.serviceMetadata = blockchain.DefaultServiceMetaData() 
            return components.serviceMetadata
        }
      
        log.Panic("Blockchain is disabled and no service metadata file is passed in configuration")
}
```

Reply:
log.Panic("Blockchain is disabled and no service metadata file is passed in configuration") seems to indicate that if the block chain is disabled, then file is mandatory  ? why enforce this when we just need 2 details ( Service Type  and Encoding ) ?

Reply:
Thinking more about this, lets make it mandatory for the service metadata file to be provided by the user. This ensures that if the developer disables the blockchain mode by mistake there is still an explicit step to be performed for the daemon to run. I would request that we clearly display the error message in the log file and on the console about whats happening.